src/base/misc/disks.c

```diff
@@ -929,163 +916,100 @@ static void image_setup(struct disk *dp)
   }
 }
 
-/* XXX - relies upon a file of SECTOR_SIZE in PARTITION_PATH that which
- *       contains the MBR (first sector) of the drive (i.e. /dev/hda).
- *       only works for /dev/hda1 right now, and only allows one
- *       partition, which must begin at head 1, sec 1, cyl 0 (i.e. linear
- *       sector 2, the second one).
- *       Also, it eats memory proportional to the number of sectors before
- *       the start of the partition.
- */
-
 static void partition_auto(struct disk *dp)
 {
+  unsigned char sect[0x200];
+  struct on_disk_vbr *vbr = (struct on_disk_vbr *)sect;
```

Comment:
Breaks strict aliasing.
I would suggest to read directly
into struct vbr instead of using
a pointer.
Not insisting for now though.
Doesn't this add a warning for
a release build?
Perhaps just disabling strict
aliasing would be an adequate
answer.

Reply:
> I would suggest to read directly
into struct vbr instead of using
a pointer.

Okay will do.

> Doesn't this add a warning for
a release build?

To be honest I never do a release build, only development.